sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -200,6 +219,30 @@ export class Receiver {
     return result;
   }
 
+  private getCheckpoint(): number | undefined {
+    if (this._streamingReceiver || this._batchingReceiver) {
+      let lastBatchingReceiverSequenceNum: number = -1;
+      let lastStreamingReceiverSequenceNum: number = -1;
+      if (this._batchingReceiver) {
+        lastBatchingReceiverSequenceNum = this._batchingReceiver.checkpoint;
+      }
+      if (this._streamingReceiver) {
+        lastStreamingReceiverSequenceNum = this._streamingReceiver.checkpoint;
+      }
+      return Math.max(lastStreamingReceiverSequenceNum, lastBatchingReceiverSequenceNum);
+    }
+    return;
```

Comment:
Generally, if the entire section inside a function is wrapped around an `if`, flipping the `if` condition makes the code cleaner to read due to the reduced nesting. For example:
```javascript
if (!this_streamingReceiver && !this._batchingReceiver) {
   return;
}
// Rest of your code
```

Reply:
In the current code there is no nesting, these is only one if condition.

I got your point but in this case I think it is more clearer if we say  like -
if there is any receiver then return the last max sequenceNumber otherwise just return.

```javascript
getcheckpoint(): number | undefined {
If(this_streamingReceiver || this._batchingReceiver){
//code
return max(lastStreamingReceiverSequnenceNumber, lastBatchingReceiverSequnenceNumber)
}
return;
}
```